spec/features/date_selection_spec.rb

```diff
@@ -0,0 +1,149 @@
+  let(:metrics) { %w[pageviews unique_pageviews number_of_internal_searches satisfaction_score] }
+
+  before do
+    initial_page_stub
+  end
+
+  it 'renders data for the last 30 days if no date period is selected' do
+    date_range = build(:date_range, :last_30_days)
+    visit '/metrics/base/path'
+    expect_default_date_period_metrics_to_be_displayed(date_range.from.to_date, date_range.to.to_date)
+  end
+
+  it 'renders data for the last 30 days when `last 30 days` is selected' do
+    date_range = build(:date_range, :last_30_days)
+    stub_response_for_date_range(date_range.from.to_date, date_range.to.to_date)
+    visit_page_and_filter_by_date_range('last-30-days')
+    expect_metrics_for_each_date_to_be_correct(date_range.from.to_date, date_range.to.to_date)
+  end
+
+  it 'renders data for the previous month when `last month` is selected' do
+    date_range = build(:date_range, :last_month)
+    stub_response_for_date_range(date_range.from.to_date, date_range.to.to_date)
+    visit_page_and_filter_by_date_range('last-month')
+    expect_metrics_for_each_date_to_be_correct(date_range.from.to_date, date_range.to.to_date)
+  end
+
+  it 'renders data for the last 3 months when `last 3 months` is selected' do
+    date_range = build(:date_range, :last_3_months)
+    stub_response_for_date_range(date_range.from.to_date, date_range.to.to_date)
+    visit_page_and_filter_by_date_range('last-3-months')
+    expect_metrics_for_each_date_to_be_correct(date_range.from.to_date, date_range.to.to_date)
+  end
+
+  it 'renders data for the last 6 months when `last6 months` is selected' do
+    date_range = build(:date_range, :last_6_months)
+    stub_response_for_date_range(date_range.from.to_date, date_range.to.to_date)
+    visit_page_and_filter_by_date_range('last-6-months')
+    expect_metrics_for_each_date_to_be_correct(date_range.from.to_date, date_range.to.to_date)
+  end
+
+  it 'renders data for the last year when `last year` is selected' do
+    date_range = build(:date_range, :last_1_year)
+    stub_response_for_date_range(date_range.from.to_date, date_range.to.to_date)
+    visit_page_and_filter_by_date_range('last-1-year')
+    expect_metrics_for_each_date_to_be_correct(date_range.from.to_date, date_range.to.to_date)
+  end
+
+  it 'renders data for the last 2 years when `last 2 years` is selected' do
+    date_range = build(:date_range, :last_2_years)
+    stub_response_for_date_range(date_range.from.to_date, date_range.to.to_date)
+    visit_page_and_filter_by_date_range('last-2-years')
+    expect_metrics_for_each_date_to_be_correct(date_range.from.to_date, date_range.to.to_date)
+  end
+
+  def initial_page_stub
+    from = Time.zone.today - 30.days
+    to = Time.zone.today
+    content_data_api_has_metric(base_path: 'base/path',
+                                from: from,
+                                to: to,
+                                metrics: metrics)
+    content_data_api_has_timeseries(base_path: 'base/path',
```

Comment:
It seems to me that the `initial_page_stub` is a way to populate some data so we 
stub the requests and don't invoke the real services. If this is the case, I would not
be too explicit here. 
If we need these fixtures, could you please explain why? I might be missing something.
though.

Reply:
No, you are correct! I have left in a payload in this example as we need it to be different from the default payload in order to differentiate between the initial payload received when we first load the page and the payload we expect back after we have filter (i think!).